include/dev/HallSensor.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <EVT/io/GPIO.hpp>
+
+namespace IO = EVT::core::IO;
+
+namespace WSS::DEV {
+
+class HallSensor {
+
+public:
+    enum WheelSpeedState {
+        STOP = 0,        // First pulse or wheel is dead
+        INITIALIZING = 1,// Setting speed based on first reading
+        MAINTAIN = 2,    // Wheel is spinning at a constant speed or speeding up
```

Comment:
```suggestion
    enum class WheelSpeedState {
        STOP,        // First pulse or wheel is dead
        INITIALIZING,// Setting speed based on first reading
        MAINTAIN,    // Wheel is spinning at a constant speed or speeding up
```
You should only use `enum` when you need to use the numerical representation for something. Otherwise, you should always use `enum class`